### PR TITLE
adjustments needed for gas-water system

### PR DESCRIPTION
**Pull request by plgbrts**
fixes to allow for gas-water system. This PR is related to https://github.com/OPM/opm-models/pull/665

### CODE REVIEW

File: opm/simulators/wells/StandardWellEval.cpp [hunk 1]
```diff
@@ -90,19 +90,19 @@ relaxationFactorFractionsProducer(const std::vector<double>& primary_variables,
     double relaxation_factor = 1.0;
 
     if (FluidSystem::numActivePhases() > 1) {
-        if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
+        if (has_wfrac_variable) {
```
Comment (akva2) [inline opm/simulators/wells/StandardWellEval.cpp:93]: please use if constexpr for purely compile time conditionals. micro-optim which allows for code elimination.

Comment (plgbrts) [reply opm/simulators/wells/StandardWellEval.cpp]: changed `if (...) {` to ` if constexpr (...) { `

**Comment by totto82** [comment]
Thanks for the contribution. The code looks good to me. Are these PRs sufficient to run gas-water or is it a step on way.

**Comment by plgbrts** [comment]
With this PR (#3454) and
-opm-models PR #665 (not merged yet)
-opm-simulators PR #3445 (merged)
-opm-material PR #446 (not merged yet)
a gas-water case can run. I will make a PR for a GAS-WATER test case.

**Comment by totto82** [comment]
I suggest you add the newly merged 2p gas-water case in the jenkins test suite so we don't loose the functionality again.

**Comment by goncalvesmachadoc** [comment]
He is about to push that. :)

**Comment by OPMUSER** [comment]
Although the test case runs because the model users Enumeration Initialization to initialize the model. If you use the standard Equilibrium Initialization the model fails to initialize because it is looking for an oil phase:

```
Program threw an exception: Could not initialize the problem: Phase pressure function for "OIL" most not be null
Failed: 2882
```

Test deck attached.

[GASW-BASE01-R01.zip](https://github.com/OPM/opm-simulators/files/7119785/GASW-BASE01-R01.zip)

**Comment by goncalvesmachadoc** [comment]
Yes, that's a known issue. Equilibration for initialization in OPM only works when there is oil in the model.

**Comment by OPMUSER** [comment]
@goncalvesmachadoc thanks for the clarification.

**Pull request merged**